flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/xa/JdbcXaSinkFunctionState.java

```diff
@@ -28,6 +30,7 @@
 
 /** Thread-safe (assuming immutable {@link Xid} implementation). */
 @ThreadSafe
+@PublicEvolving
```

Comment:
How about removing the `@PublicEvolving` of `XaSinkStateHandler`? Because it seems that it should not be a user interface (only used in `JdbcXaSinkFunction`, and `JdbcXaSinkFunction` is `@Internal`).
Once `XaSinkStateHandler` is not `@PublicEvolving`, the change here will be not needed.

cc @rkhachatryan

Reply:
Hi @wanglijie95,
Its a param of the constructor of JdbcXaSinkFunction (line 199), because of that I set it to PublicEvolving.
All params should be public no?

Reply:
But the `JdbcXaSinkFunction` is `@Internal`, which means the users are not expectd to use it directly(call any constructor of it)

Reply:
yes true, changed